test/e2e/federated-service.go

```diff
@@ -0,0 +1,209 @@
+			clusterClientSets = append(clusterClientSets, clset)
+		}
+	})
+
+	It("should be able to discover a federated service", func() {
+		framework.SkipUnlessFederated(f.Client)
+
+		createService(f.FederationClientset, clusterClientSets, f.Namespace.Name)
+
+		discoverService(f)
+	})
+
+	It("should be able to discover a non-local federated service", func() {
+		framework.SkipUnlessFederated(f.Client)
+
+		createService(f.FederationClientset, clusterClientSets, f.Namespace.Name)
+
+		// Delete a federated service shard in the default e2e Kubernetes cluster.
+		err := f.Clientset_1_3.Core().Services(f.Namespace.Name).Delete(FederatedServiceName, &api.DeleteOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		waitForFederatedServiceShard(f.Clientset_1_3, f.Namespace.Name, nil, 0)
+
+		discoverService(f)
+	})
+})
+
+func waitForFederatedServiceShard(cs *release_1_3.Clientset, namespace string, service *api.Service, numSvcs int) {
+	By("Fetching a federated service shard")
+	var clSvcList *v1.ServiceList
+	for t := time.Now(); time.Since(t) < FederatedServiceTimeout; time.Sleep(framework.Poll) {
+		var err error
+		clSvcList, err = cs.Core().Services(namespace).List(api.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		n := len(clSvcList.Items)
+		if n == numSvcs {
+			break
+		}
+		framework.Logf("%d services found, trying again in %s", n, framework.Poll)
+	}
+
+	Expect(len(clSvcList.Items)).To(Equal(numSvcs))
+	if numSvcs > 0 {
+		// Renaming for clarity/readability
+		clSvc := clSvcList.Items[0]
+		Expect(clSvc.Name).To(Equal(service.Name))
+		Expect(clSvc.Spec).To(Equal(service.Spec))
+	}
+}
+
+func createService(fcs *federation_internalclientset.Clientset, clusterClientSets []*release_1_3.Clientset, ns string) {
+	By("Creating a federated service")
+	labels := map[string]string{
+		"foo": "bar",
+	}
+
+	svc1port := "svc1"
+	svc2port := "svc2"
+
+	service := &api.Service{
+		ObjectMeta: api.ObjectMeta{
+			Name: FederatedServiceName,
+		},
+		Spec: api.ServiceSpec{
+			Selector: labels,
+			Ports: []api.ServicePort{
+				{
+					Name:       "portname1",
+					Port:       80,
+					TargetPort: intstr.FromString(svc1port),
+				},
+				{
+					Name:       "portname2",
+					Port:       81,
+					TargetPort: intstr.FromString(svc2port),
+				},
+			},
+		},
+	}
+	_, err := fcs.Core().Services(ns).Create(service)
+	Expect(err).NotTo(HaveOccurred())
+	for _, cs := range clusterClientSets {
+		waitForFederatedServiceShard(cs, ns, service, 1)
+	}
+}
+
+func discoverService(f *framework.Framework) {
+	pod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Name:   FederatedServicePod,
+			Labels: map[string]string{"name": FederatedServicePod},
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{
+				{
+					Name:    "federated-service-discovery-container",
+					Image:   "gcr.io/google_containers/busybox:1.24",
+					Command: []string{"sh", "-c", "nslookup", FederatedServiceName},
```

Comment:
I am not sure if I understand the question correctly. Can you please rephrase/elaborate?